src/core/core.scale.js

```diff
@@ -668,16 +676,11 @@ module.exports = function(Chart) {
 						context.lineTo(itemToDraw.tx2, itemToDraw.ty2);
 					}
 
-					if (gridLines.drawOnChartArea) {
-						context.moveTo(itemToDraw.x1, itemToDraw.y1);
-						context.lineTo(itemToDraw.x2, itemToDraw.y2);
-					}
-
 					context.stroke();
 					context.restore();
 				}
 
-				if (optionTicks.display) {
```

Comment:
items with null or undefined label aren't pushed into the array at all, thanks to **[this](https://github.com/chartjs/Chart.js/blob/master/src/core/core.scale.js#L551-L554)** statement.
The blank labels don't need to be drawn tho. The question is if they should be skipped the same way as when it's null(its ticks and gridLines wont be drawn) or in the statement you linked so there will be a blank tick, but the label itself wont be drawn.